backend_plugin/src/main/evaluators/AddImportEvaluator.java

```diff
@@ -0,0 +1,104 @@
+package main.evaluators;
+
+import java.util.Iterator;
+
+import org.eclipse.jface.text.BadLocationException;
+import org.eclipse.jface.text.DocumentEvent;
+import org.eclipse.jface.text.source.Annotation;
+import org.eclipse.jface.text.source.IAnnotationModel;
+
+public class AddImportEvaluator extends FeatureEvaluator {
+	
+	private boolean unresolvedVariablesExist;
+	private DocumentEvent lastDocumentEvent;
```

Comment:
The `lastDocumentEvent` maybe should live in the `Evaluator` abstract class rather than in the `AddImportEvaluator`. I think it's fine for now because no other evaluators keep track o the last document event, but it's something to keep in mind if any others of them start tracking this.